tests/test_evpn_l3_vxlan.py

```diff
@@ -0,0 +1,1726 @@
+        create_vrf_routes(dvs, "2002::8/64", 'Vrf-RED', '8.8.8.8', "Vlan100", "00:22:22:22:22:22", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "2002::8/64", 'Vrf-RED', '8.8.8.8', tunnel_name, "00:22:22:22:22:22", '1000', 1)
+        print "\tPass"
+
+        print "\tTest Previous Tunnel Nexthop 7.7.7.7 is removed"
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv6 Route and Tunnel Nexthop 8.8.8.8 Delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+        delete_vrf_routes(dvs, "2002::8/64", 'Vrf-RED')
+        vxlan_obj.check_del_tunnel_nexthop(dvs, 'Vrf-RED', '8.8.8.8', tunnel_name, "00:22:22:22:22:22", '1000')
+        vxlan_obj.check_del_vrf_routes(dvs, "2002::8/64", 'Vrf-RED')
+        print "\tPass"
+
+        print "\n\nTest VRF IPv6 Route with ECMP Tunnel Nexthop Add and delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+
+        ecmp_nexthop_attr = [
+            ("nexthop", "7.7.7.7,8.8.8.8"),
+            ("ifname", "Vlan100,Vlan100"),
+            ("vni_label", "1000,1000"),
+            ("router_mac", "00:11:11:11:11:11,00:22:22:22:22:22"),
+        ]
+
+        print "\tTest VRF IPv6 Route with ECMP Tunnel Nexthop [7.7.7.7 , 8.8.8.8] Add"
+        create_vrf_routes_ecmp(dvs, "2002::8/64", 'Vrf-RED', ecmp_nexthop_attr)
+
+        nh_count = 2
+        ecmp_nhid_list = vxlan_obj.check_vrf_routes_ecmp(dvs, "2002::8/64", 'Vrf-RED', tunnel_name, nh_count)
+        assert nh_count == len(ecmp_nhid_list)
+        vxlan_obj.check_add_tunnel_nexthop(dvs, ecmp_nhid_list[0], '7.7.7.7', tunnel_name, '00:11:11:11:11:11', '1000')
+        vxlan_obj.check_add_tunnel_nexthop(dvs, ecmp_nhid_list[1], '8.8.8.8', tunnel_name, '00:22:22:22:22:22', '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv6 Route with ECMP Tunnel Nexthop [7.7.7.7 , 8.8.8.8] Delete"
+        vxlan_obj.fetch_exist_entries(dvs)
+        delete_vrf_routes(dvs, "2002::8/64", 'Vrf-RED')
+        check_deleted_object(self.adb, vxlan_obj.ASIC_NEXT_HOP, ecmp_nhid_list[0])
+        check_deleted_object(self.adb, vxlan_obj.ASIC_NEXT_HOP, ecmp_nhid_list[1])
+        
+        vxlan_obj.check_vrf_routes_ecmp_nexthop_grp_del(dvs, 2)
+        vxlan_obj.check_del_vrf_routes(dvs, "2002::8/64", 'Vrf-RED')
+        print "\tPass"
+
+        print "\n\nTest VRF IPv6 Route with Tunnel Nexthop update from non-ECMP to ECMP"
+        print "\tTest VRF IPv6 Route with Tunnel Nexthop 7.7.7.7 Add"
+        vxlan_obj.fetch_exist_entries(dvs)
+        create_vrf_routes(dvs, "2002::8/64", 'Vrf-RED', '7.7.7.7', "Vlan100", "00:11:11:11:11:11", '1000')
+        vxlan_obj.check_vrf_routes(dvs, "2002::8/64", 'Vrf-RED', '7.7.7.7', tunnel_name, "00:11:11:11:11:11", '1000')
+        print "\tPass"
+
+        print "\tTest VRF IPv4 Route with ECMP Tunnel Nexthop [7.7.7.7 , 8.8.8.8] Udpate"
+        ecmp_nexthop_attr = [
```

Comment:
If we can reuse the earlier declaration of "ecmp_nexthop_attr " . Please use it.

Reply:
ok. Will do.